### PR TITLE
Ensure zarr based DataArrays are not loaded into memory

### PR DESCRIPTION
**Pull request by philippjfr**
Fixes https://github.com/holoviz/hvplot/issues/706

### CODE REVIEW

**Comment by jbednar** [comment]
Thanks. Is there a way to demonstrate whether this fixes the issue?

**Comment by maximlt** [comment]
I've added a test to check that, when the data is "symmetric" but is "not in memory", `_process_symmetric` returns False early to avoid computing `nanquantile` on the whole data which would eagerly loads it if it was not yet already done.

This PR is actually a bigger change than I thought. For instance, opening a local NetCDF file as such `import xarray as xr; ds = xr.open_dataset('data.nc')` will not fully load it "in memory", i.e. `ds.variable._in_memory` will be `False` (this is what I've done to build the test by the way). If the `ds.variable` data happened to be symmetric then it will no longer be displayed with a diverging color map after this PR, at least not without passing directly `symmetric=True`.

Thoughts on this @jbednar, @philippjfr ?

**Comment by jbednar** [comment]
@rabernat, do you have any suggestions?

File: hvplot/converter.py [hunk 1]
```diff
@@ -599,9 +599,9 @@ def _process_symmetric(self, symmetric, clim, check_symmetric_max):
 
         if is_xarray(self.data):
             # chunks mean it's lazily loaded; nanquantile will eagerly load
-            if self.data.chunks:
-                return False
             data = self.data[self.z]
+            if not getattr(data, '_in_memory', True) or data.chunks:
+                return False
```
Comment (rabernat) [inline hvplot/converter.py:604]: This seems reasonable to me insofar as it treats non-in-memory data the same as chunked data

**Review by rabernat — APPROVED** [review]
Everything here looks good to me!

**Comment by rabernat** [comment]
> If the `ds.variable` data happened to be symmetric then it will no longer be displayed with a diverging color map after this PR, at least not without passing directly `symmetric=True`.

Is it possible for this behavior to be dependent on whether this is a "dynamic" plot or not? Or some way for the user to explicitly indicate that they do not want or expect the data to be loaded eagerly.

**Comment by maximlt** [comment]
This is indeed a change in behavior, but we believe that most hvPlot users are more likely to be in a space where the new behavior is to their advantage, such as climate/earth scientists that use hvPlot to display very large datasets and for who loading the full data is just plain wrong.

**Pull request merged**